postgres-appliance/scripts/configure_spilo.py

```diff
@@ -691,7 +691,8 @@ def write_wale_environment(placeholders, prefix, overwrite):
     s3_names = ['WALE_S3_PREFIX', 'WALG_S3_PREFIX', 'AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY',
                 'WALE_S3_ENDPOINT', 'AWS_ENDPOINT', 'AWS_REGION', 'AWS_INSTANCE_PROFILE',
                 'WALG_S3_SSE_KMS_ID', 'WALG_S3_SSE', 'WALG_DISABLE_S3_SSE', 'AWS_S3_FORCE_PATH_STYLE']
-    azure_names = ['WALG_AZ_PREFIX', 'AZURE_STORAGE_ACCOUNT', 'AZURE_STORAGE_ACCESS_KEY']
+    azure_names = ['AZURE_STORAGE_ACCESS_KEY', 'AZURE_STORAGE_ACCOUNT', 'AZURE_STORAGE_SAS_TOKEN',
+                   'WALG_AZURE_BUFFER_SIZE', 'WALG_AZURE_MAX_BUFFERS', 'WALG_AZ_PREFIX']
```

Comment:
The `WALG_AZ_PREFIX` must be placed first in the list.

```suggestion
    azure_names = ['WALG_AZ_PREFIX', 'AZURE_STORAGE_ACCOUNT', 'AZURE_STORAGE_ACCESS_KEY',
                   'AZURE_STORAGE_SAS_TOKEN', 'WALG_AZURE_BUFFER_SIZE', 'WALG_AZURE_MAX_BUFFERS']
```

Reply:
Ah, I wasn't aware there was a required ordering - although I'm curious if there is a specific reason?

Reply:
The `WAL{E,G}_??_PREFIX` path could be calculated based on `WAL_??_BUCKET`, `SCOPE`, `PGVERSION` and a few more less important env variables:
https://github.com/zalando/spilo/blob/918b3d6d93be4638a54976a5c150dd846fd7abdf/postgres-appliance/scripts/configure_spilo.py#L761-L766

It makes an assumption that the `WAL{E,G}_??_PREFIX` is first in the list to calculate the URL scheme.
Although, in case of Azure the scheme seems to be `azure://`, not `az://` like in wal-e.

In order to make it work automatically, maybe changing of https://github.com/zalando/spilo/blob/918b3d6d93be4638a54976a5c150dd846fd7abdf/postgres-appliance/scripts/configure_spilo.py#L768 would also be required.